File: 020-valid-parentheses/isValid.cpp
#include <iostream>
#include <stack>

using namespace std;

bool isValid(string s) {
    stack <int> myStack;
    for (int i = 0; i < s.length(); i++) {
        char value = s.at(i);
        cout<< value<<endl;
        if (value == '(' || value == '{' || value == '[') {
            myStack.push(value);
        } else {
            cout<<myStack.size()<<endl;
            char left = myStack.top();
            cout<<myStack.size()<<endl;
            char right = s.at(i);
            if (!((left == '(' && right == ')') || (left == '{' && right == '}') || (left == '[' && right == ']'))) {
                return false;
            }else{
                myStack.pop();
            }
        }
    }
    if (myStack.size() == 0) {
        return true;
    } else {
        return false;
    }
}

int main(){
    string s = "{[{}]}";
    cout<<isValid(s)<<endl;
    return 0;
}

File: 020-valid-parentheses/isValid.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    var stack = [];
    for (var i = 0; i < s.length; i++) {
        var value = s.charAt(i);
        if (value == '(' || value == '{' || value == '[') {
            stack.push(value);
        } else {
            var left = stack.pop();
            var right = s.charAt(i);
            if (!((left == '(' && right == ')') || (left == '{' && right == '}') || (left == '[' && right == ']'))) {
                return false;
            }
        }
    }
    if (stack.length === 0) {
        return true;
    } else {
        return false;
    }
};
console.log(isValid('[()]'));
